From fa74fd7afa5040e8694e106231fe6aeba64b06b5 Mon Sep 17 00:00:00 2001
From: Andreas Mohr <and@gmx.li>
Date: Mon, 7 Dec 2020 12:42:26 +0000
Subject: [PATCH] (extfs.c) fix uninitialized scalar variable

Fix uninitialized scalar variable

Found by Coverity

coverity id #32627
coverity id #313665
covertiy id #313669

Signed-off-by: Andreas Mohr <and@gmx.li>
---
 src/vfs/extfs/extfs.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/vfs/extfs/extfs.c b/src/vfs/extfs/extfs.c
index 7ddbc7a80..da263b004 100644
--- a/src/vfs/extfs/extfs.c
+++ b/src/vfs/extfs/extfs.c
@@ -200,6 +200,7 @@ extfs_generate_entry (struct extfs_super_t *archive, const char *name, struct vf
     struct vfs_s_inode *inode;
     struct vfs_s_entry *entry;
 
+    memset(&st, 0, sizeof(struct stat));
     st.st_ino = VFS_SUPER (archive)->ino_usage++;
     st.st_dev = archive->rdev;
     myumask = umask (022);
@@ -408,6 +409,8 @@ extfs_open_archive (int fstype, const char *name, struct extfs_super_t **pparc)
     vfs_path_t *local_name_vpath = NULL;
     vfs_path_t *name_vpath;
 
+    memset(&mystat, 0, sizeof(struct stat));
+
     name_vpath = vfs_path_from_str (name);
     info = &g_array_index (extfs_plugins, extfs_plugin_info_t, fstype);
 
@@ -562,6 +565,7 @@ extfs_read_archive (FILE * extfsd, struct extfs_super_t *current_archive)
                 else
                 {
                     struct stat st;
+                    memset(&st, 0, sizeof(struct stat));
 
                     st.st_ino = super->ino_usage++;
                     st.st_nlink = 1;

